src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm

```diff
@@ -88,6 +90,22 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
     return "@";
 }
 
+const Metadata::InterfaceMeta* ObjCConstructorBase::metadata() {
+    if (this->_metadata == nullptr) {
+        JSObject* proto = this->_prototype.get();
+        ObjCPrototype* objcPrototype = nullptr;
+        while (proto && !(objcPrototype = jsDynamicCast<ObjCPrototype*>(proto))) {
+            proto = (proto->prototype().isCell()) ? jsDynamicCast<ObjCPrototype*>(proto->prototype().asCell()) : nullptr;
```

Comment:
I think this is best written as `proto = jsDynamicCast<JSObject*>(proto->prototype());` since `jsDynamicCast` is already checking for cells and stuff.